src/main/java/appeng/helpers/PatternHelper.java

```diff
@@ -85,7 +85,13 @@ public PatternHelper( final ItemStack is, final World w )
 
 		for( int x = 0; x < inTag.tagCount(); x++ )
 		{
-			final ItemStack gs = new ItemStack( inTag.getCompoundTagAt( x ) );
+			NBTTagCompound ingredient = inTag.getCompoundTagAt( x );
+			final ItemStack gs = new ItemStack( ingredient );
+
+			if( !ingredient.hasNoTags() && gs.isEmpty() )
+			{
+				throw new IllegalArgumentException( "No pattern here!" );
```

Comment:
Sorry, I wanted to answer you, asking for the further direction as you concern totally made sense to me, but I didn't find the time in the last days. Thanks for merging anyways :)